### PR TITLE
fix/core_tests: make balaced routing do what it says it does

### PR DESCRIPTION
**Pull request by dhardy**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @afck

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/core_tests.rs [hunk 1]
```diff
@@ -264,18 +264,17 @@ fn poll_all(nodes: &mut [TestNode], clients: &mut [TestClient]) -> bool {
     loop {
         let mut n = false;
         if BALANCED_POLLING {
-            nodes.iter_mut().foreach(|node| n = n || node.inner.poll());
+            // handle at most one event from each node:
```
Comment (Fraser999) [inline src/core_tests.rs]: I think [`node.inner.poll()`](https://github.com/Fraser999/routing/blob/master/src/core_tests.rs#L157-L166) is handling _all_ events from each node isn't it?

File: src/core_tests.rs [hunk 2]
```diff
@@ -264,18 +264,17 @@ fn poll_all(nodes: &mut [TestNode], clients: &mut [TestClient]) -> bool {
     loop {
         let mut n = false;
```
Comment (Fraser999) [inline src/core_tests.rs]: Since we're in here anyway, could `n` be changed to something more meaningful, e.g. `events_occurred` or something?

**Comment by dhardy** [comment]
This appears to fail `core_tests::multiple_joining_nodes`.

**Comment by afck** [comment]
It probably makes sense to disable `multiple_joining_nodes` until we can rule out contradictory quora again.

**Pull request merged**